.github/workflows/bench.yml

```diff
@@ -0,0 +1,42 @@
+name: Benchmarks
```

Comment:
This workflow uses [benchstat](https://pkg.go.dev/golang.org/x/perf/cmd/benchstat#section-readme) a common go tool for comparing different benchmarks results.
In our case, the workflow will compare the `master` branch benchmark's result with the ones of a new PR which is IMO very useful to let us understand if we have introduced a performance regression.
In order to make this properly working, bench tests should be written when we consider it's the case.